engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -70,11 +70,27 @@ case class FixedValuesValidator(possibleValues: List[FixedExpressionValue]) exte
   }
 }
 
-case object LiteralIntValidator extends ParameterValidator {
-
+case class RegExpParameterValidator(pattern: String, message: String, description: String) extends ParameterValidator {
   override def isValid(paramName: String, value: String, label: Option[String])
                       (implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] = {
-
-    if (Try(value.toInt).isSuccess) valid(Unit) else invalid(ProcessCompilationError.InvalidLiteralIntValue(paramName, label, value))
+    if (Pattern.compile(pattern).matcher(value).matches()) valid(Unit) else invalid(NotMatchParameter(message, description, paramName, nodeId.id))
   }
 }
+
+case object LiteralParameterValidator {
+  val integerValidator: RegExpParameterValidator = RegExpParameterValidator(
+    "^-?[0-9]+$",
```

Comment:
If we want to be really correct we would want to validate also length of this literal :P

Reply:
I've added checking isBlank at RegExp ;)

Reply:
but I'm not talking about too short - I'm talking about too long number and checking for integer overflow :D

Reply:
Hmm this should be catched on next level validation - I mean on typing? :p

Reply:
@mproch has right that for ints regexp are not the best choice  - let's say someone provide 111111111111111111111111111111111. The correct solution should be checking if `Try(expr.toInt).isSucces`... but regexp can be functional for handling `@Pattern` annotation :)